File: 3483-unique-3-digit-even-numbers/3483-unique-3-digit-even-numbers.js
/**
 * @param {number[]} digits
 * @return {number}
 */
var totalNumbers = function(digits) {
    const freq = new Array(10).fill(0);

    for (const digit of digits) {
        freq[digit]++;
    }
    
    let count = 0;

    for (let num = 100; num <= 999; num++) {
        // Skip odd numbers
        if ((num & 1) !== 0) {
            continue;
        }
        
        const a = Math.floor(num / 100);
        const b = Math.floor(num / 10) % 10;
        const c = num % 10;

        const used = new Array(10).fill(0);
        used[a]++;
        used[b]++;
        used[c]++;

        let possible = true;

        for (let d = 0; d <= 9; d++) {
            if (used[d] > freq[d]) {
                possible = false;
                break;
            }
        }

        if (possible) {
            count++;
        }
    }

    return count;
};

File: 3483-unique-3-digit-even-numbers/3483-unique-3-digit-even-numbers.py
class Solution:
    def totalNumbers(self, digits: list[int]) -> int:
        freq = [0] * 10

        for digit in digits:
            freq[digit] += 1
        
        count = 0

        for num in range(100, 1000):
            # Skip odd numbers
            if (num & 1) != 0:
                continue
            
            a = num // 100
            b = (num // 10) % 10
            c = num % 10

            used = [0] * 10
            used[a] += 1
            used[b] += 1
            used[c] += 1

            possible = True

            for d in range(10):
                if used[d] > freq[d]:
                    possible = False
                    break

            if possible:
                count += 1

        return count
